final-project/static/scripts.js

```diff
@@ -247,68 +253,88 @@ function editProduct() {
+  prodPageElements.forEach(function(value, key) {
+
+    // Create a form element.
+    var formElement = document.createElement('form');
+    // Name the form.
+    formElement.setAttribute('name', key + '-form');
 
     // On the references form, loop through all the reference items.
-    if (key === references) {
+    if (key === 'references') {
 
       // Loop through references.
       for(var i = 0; i < value.length; i++) {
 
-        // Create a form element.
-        formElement = document.createElement('form');
+        // Create a new form element.
+        var formElement = document.createElement('form');
         // Name the form.
         formElement.setAttribute('name', 'reference-form-' + i);
 
         // Create an input for the title
         var titleInput = document.createElement('input');
         // Name the input.
         titleInput.setAttribute('name', 'reference-title-input-' + i);
+        // Add placeholder text with current title.
+        titleInput.setAttribute('placeholder', value[i].firstChild.innerText);
 
         // Create an input for the link.
         var linkInput = document.createElement('input');
         // Name the input.
-        titleInput.setAttribute('name', 'reference-link-input-' + i);
+        linkInput.setAttribute('name', 'reference-link-input-' + i);
+        linkInput.setAttribute('placeholder', value[i].firstChild.href);
 
         // Append an input for the title.
         formElement.appendChild(titleInput);
         // Append an input for the link.
         formElement.appendChild(linkInput);
         // Append the form to the product page element.
-        value.appendChild(formElement);
+        value[i].appendChild(formElement);
       }
 
-    // On all other forms append a single input.
-    } else {
+    // On the image form, use the URL as a placeholder.
+    } else if (key === 'image') {
+
+      // Create input element.
+      var inputElement = document.createElement('input');
+      // Name the element.
+      inputElement.setAttribute('name', key + '-input');
+      // Provide placeholder text of current src.
+      inputElement.setAttribute('placeholder', value.src)
 
-      // Create a form element.
-      formElement = document.createElement('form');
-      // Name the form.
-      formElement.setAttribute('name', key + '-form');
+      // Append the input element to the form element.
+      formElement.appendChild(inputElement);
+      // Append the form to the product page element.
+      value.appendChild(formElement);
+
+    // Otherwise, on all other elements use the innerText as a placeholder.
+    } else {
 
       // Create input element.
-      inputElement = document.createElement('input');
+      var inputElement = document.createElement('input');
       // Name the element.
       inputElement.setAttribute('name', key + '-input');
+      // Provide inner text of current placeholder.
+      inputElement.setAttribute('placeholder', value.innerText)
```

Comment:
(nit) It's good to be consistent with using or not using semi-colons.

Reply:
Definitely. Good catch. I think I can use JSLint to scan the whole file for stuff like this. Will do.

Reply:
This is completed in the https://github.com/jwq1/CS50x_jwq1/pull/60/commits/7cca3cddd72cb4cf4ca2c0a09d8fd2a7e0dc249b commit below